subsys/random/rand32_ctr_drbg.c

```diff
@@ -28,7 +28,7 @@
 
 static K_SEM_DEFINE(state_sem, 1, 1);
 
-static const struct device *entropy_driver;
+static const struct device *entropy_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_entropy));
```

Comment:
unfortunately I missed this "chime" request... there is a bug that was introduced with this change because the initialization function is only called based on whether this entropy_dev object has been assigned (see line 105 below).